### ISSUE TITLE
Get-AzInsightsPrivateLinkScope from -ResourceId does not return anything

### ISSUE THREAD

**Issue by brianstringfellow** [Azure/azure-powershell#22568]
### Description

When I run `Get-AzInsightsPrivateLinkScope -ResourceId $resourceId` the cmdlet doesn't return anything. As a sanity check, I ran the cmdlet with the alternate parameter set with a resource group and that worked.

```
PS>$amplsRg = Get-AzInsightsPrivateLinkScope -ResourceGroupName 'My-Resource-Group' -Name 'mypls00'
>> $amplsRg -eq $null
>> $amplsRg.Id
>> $amplsId = Get-AzInsightsPrivateLinkScope -ResourceId $amplsRg.Id
>> $amplsId -eq $null
>> $amplsId.Id
WARNING: 16:34:23 - *** The namespace for all the model classes will change from Microsoft.Azure.Management.Monitor.Management.Models to Microsoft.Azure.Management.Monitor.Models in future releases.
WARNING: 16:34:23 - *** The namespace for output classes will be uniform for all classes in future releases to make it independent of modifications in the model classes.
False
/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/My-Resource-Group/providers/microsoft.insights/privatelinkscopes/mypls00
WARNING: 16:34:24 - *** The namespace for all the model classes will change from Microsoft.Azure.Management.Monitor.Management.Models to Microsoft.Azure.Management.Monitor.Models in future releases.
WARNING: 16:34:24 - *** The namespace for output classes will be uniform for all classes in future releases to make it independent of modifications in the model classes.
True
```

As an aside, I thought it was odd that there is not a test in the following file for the ResourceId parameter set.
https://github.com/Azure/azure-powershell/blob/main/src/Monitor/Monitor.Test/ScenarioTests/PrivateLinkScopeTests.ps1

### Issue script & Debug output

```PowerShell
PS>Get-AzInsightsPrivateLinkScope -ResourceId '/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/My-Resource-Group/providers/microsoft.insights/privateLinkScopes/mypls00'
DEBUG: 16:28:02 - GetAzureInsightsPrivateLinkScope begin processing with ParameterSet 'ByResourceIdParameterSet'.
DEBUG: 16:28:02 - using account id 'notme@company.com'...
DEBUG: 16:28:02 - [ConfigManager] Got nothing from [DisplayBreakingChangeWarning], Module = [], Cmdlet = []. Returning default value [True].
WARNING: 16:28:02 - *** The namespace for all the model classes will change from Microsoft.Azure.Management.Monitor.Management.Models to Microsoft.Azure.Management.Monitor.Models in future releases.
WARNING: 16:28:02 - *** The namespace for output classes will be uniform for all classes in future releases to make it independent of modifications in the model classes.
DEBUG: 16:28:02 - [ConfigManager] Got nothing from [DisplayRegionIdentified], Module = [], Cmdlet = []. Returning default value [True].
DEBUG: AzureQoSEvent:  Module: Az.Monitor:4.5.0; CommandName: Get-AzInsightsPrivateLinkScope; PSVersion: 7.3.5; IsSuccess: True; Duration: 00:00:00.0016997
DEBUG: 16:28:02 - [ConfigManager] Got nothing from [EnableDataCollection], Module = [], Cmdlet = []. Returning default value [True].
DEBUG: 16:28:03 - GetAzureInsightsPrivateLinkScope end processing.
DEBUG: 16:28:03 - [ConfigManager] Got nothing from [DisplayRegionIdentified], Module = [], Cmdlet = []. Returning default value [True].
DEBUG: 16:28:03 - GetAzureRMContextCommand end processing.
```

### Environment data

```PowerShell
Name                           Value
----                           -----
PSVersion                      7.3.5
PSEdition                      Core
GitCommitId                    7.3.5
OS                             Microsoft Windows 10.0.22621
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

### Module versions

```PowerShell
ModuleType Version    PreRelease Name                                ExportedCommands
---------- -------    ---------- ----                                ----------------
Script     2.12.4                Az.Accounts                         {Add-AzEnvironment, Clear-AzConfig, Clear-AzContext, Clear-AzDefault…}
Script     4.5.0                 Az.Monitor                          {Add-AzLogProfile, Add-AzMetricAlertRule, Add-AzMetricAlertRuleV2, Add-AzWebt…
```

### Error output

```PowerShell
PS>Resolve-AzError
DEBUG: 16:25:56 - ResolveError begin processing with ParameterSet 'AnyErrorParameterSet'.
DEBUG: 16:25:56 - using account id 'notme@company.com'...
DEBUG: 16:25:56 - [ConfigManager] Got nothing from [DisplayBreakingChangeWarning], Module = [], Cmdlet = []. Returning default value [True].

   HistoryId: 1

Message        : Cannot find path 'C:\ProgramData\chocolatey\license\chocolatey.license.xml' because it does not exist.
StackTrace     :    at System.Management.Automation.LocationGlobber.ExpandMshGlobPath(String path, Boolean allowNonexistingPaths, PSDriveInfo
                 drive, ContainerCmdletProvider provider, CmdletProviderContext context)
                    at System.Management.Automation.LocationGlobber.ResolveDriveQualifiedPath(String path, CmdletProviderContext context, Boolean
                 allowNonexistingPaths, CmdletProvider& providerInstance)
                    at System.Management.Automation.LocationGlobber.GetGlobbedMonadPathsFromMonadPath(String path, Boolean allowNonexistingPaths,
                 CmdletProviderContext context, CmdletProvider& providerInstance)
                    at System.Management.Automation.LocationGlobber.GetGlobbedProviderPathsFromMonadPath(String path, Boolean
                 allowNonexistingPaths, CmdletProviderContext context, ProviderInfo& provider, CmdletProvider& providerInstance)
                    at System.Management.Automation.SessionStateInternal.GetItem(String[] paths, CmdletProviderContext context)
                    at Microsoft.PowerShell.Commands.GetItemCommand.ProcessRecord()
Exception      : System.Management.Automation.ItemNotFoundException
InvocationInfo : {Get-Item}
Line           :     $licenseFile = Get-Item -Path "$env:ChocolateyInstall\license\chocolatey.license.xml" -ErrorAction Stop

Position       : At C:\ProgramData\chocolatey\helpers\ChocolateyTabExpansion.ps1:66 char:20
                 + … censeFile = Get-Item -Path "$env:ChocolateyInstall\license\chocolatey …
                 +               ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
HistoryId      : 1



DEBUG: 16:25:56 - [ConfigManager] Got nothing from [DisplayRegionIdentified], Module = [], Cmdlet = []. Returning default value [True].
DEBUG: AzureQoSEvent:  Module: Az.Accounts:2.12.4; CommandName: Resolve-AzError; PSVersion: 7.3.5; IsSuccess: True; Duration: 00:00:00.0084382
DEBUG: 16:25:56 - [ConfigManager] Got nothing from [EnableDataCollection], Module = [], Cmdlet = []. Returning default value [True].
DEBUG: 16:25:56 - ResolveError end processing.

DEBUG: 16:25:56 - [ConfigManager] Got nothing from [DisplayRegionIdentified], Module = [], Cmdlet = []. Returning default value [True].
DEBUG: 16:25:56 - GetAzureRMContextCommand end processing.
```

**Comment by isra-fel** [comment]
Thanks for reporting!
@NoriZC could you take a look? It looks like a bug in the ResourceId parameter set. Yabo can provide you with assistance.

**Comment by NoriZC** [comment]
Resolved by #22680
